### PR TITLE
fix(sidebar-pipeline): spinner if stages undefined

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-842

Capture d’écran 2023-09-20 à 16 22 56

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

File: libs/pages/logs/environment/src/lib/ui/sidebar-pipeline/sidebar-pipeline.tsx [hunk 1]
```diff
@@ -13,16 +14,22 @@ export function SidebarPipeline({ services, versionId, serviceId, statusStages }
...
-        />
-      ))}
+      {!statusStages ? (
+        <div className="flex justify-center">
+          <LoaderSpinner className="w-6 h-6" theme="dark" />
+        </div>
+      ) : (
+        statusStages.map((currentStage: DeploymentStageWithServicesStatuses, index: number) => (
+          <SidebarPipelineItem
+            key={index}
```
Comment (ctjhoa) [inline libs/pages/logs/environment/src/lib/ui/sidebar-pipeline/sidebar-pipeline.tsx:24]: `key` should not be index, use unique identifier instead like ```key={`${serviceId}${versionId}`}```

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/pages/logs/environment/src/lib/ui/sidebar-pipeline/sidebar-pipeline.tsx (hunk 1 above)
Comment (RemiBonnet) [reply libs/pages/logs/environment/src/lib/ui/sidebar-pipeline/sidebar-pipeline.tsx:24]: I agree with this rule, but here we can't do that, we don't have a unique id on the `currentStage` and the `serviceId` +  `versionId` is the same for each Item

Comment (RemiBonnet) [reply libs/pages/logs/environment/src/lib/ui/sidebar-pipeline/sidebar-pipeline.tsx:25]: done

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**